### PR TITLE
CI various improvements

### PR DESCRIPTION
**Pull request by 0penBrain**
This PR brings some improvements to CI.

Structure:
* Cmake configure, build & install steps are factorized in a composite action for both Linux jobs

Performance:
* Logging is improved for configure, build & install steps
  * Line buffered output for stdout & stderr
  * Full log contains both stdout & stderr (previously only stdout) so errors can be looked at contextually

Other:
* Rename Ubuntu build jobs for better comprehension
* Spell checker (lint) will silently fail. Not failing the workflow anymore, but still reporting errors
* In wrapup job, deleting used artifact is allowed to fail (already happened that it failed a job)

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8463">pipeline status</a> for feature branch [PR_8463](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8463). Pipeline [775329525](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/775329525) was triggered at [94a36fd](https://github.com/FreeCAD/FreeCAD/pull/8463/commits/94a36fdc549f6ee5f254955bda063cb3d15bc2b1). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: .github/workflows/actions/linux/build/action.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# ***************************************************************************
+# *   Copyright (c) 2023 0penBrain                               *
```
Comment (luzpaz) [inline .github/workflows/actions/linux/build/action.yml:2]: Fix whitespace

File: .github/workflows/actions/linux/configure/action.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# ***************************************************************************
+# *   Copyright (c) 2023 0penBrain                               *
```
Comment (luzpaz) [inline .github/workflows/actions/linux/configure/action.yml:2]: Fix whitespace

File: .github/workflows/actions/linux/install/action.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# ***************************************************************************
+# *   Copyright (c) 2023 0penBrain                               *
```
Comment (luzpaz) [inline .github/workflows/actions/linux/install/action.yml:2]: Fix whitespace

**Review by luzpaz — CHANGES_REQUESTED** [review]

File: .github/workflows/actions/linux/build/action.yml (hunk 1 above)
Comment (0penBrain) [reply .github/workflows/actions/linux/build/action.yml:2]: See #8223

**Pull request merged**